tests/rados/test_scrub_deepscrub_timecheck.py

```diff
@@ -0,0 +1,190 @@
+            )
+            log.info(f"The osd.{osd_id} value is set to the -{capacity_iops_value}")
+        init_pool_pg_dump = rados_object.get_ceph_pg_dump(pg_id=pg_id)
+
+        init_deep_scrub_stamp = datetime.datetime.strptime(
+            init_pool_pg_dump["last_deep_scrub_stamp"], "%Y-%m-%dT%H:%M:%S.%f%z"
+        )
+        log.info(
+            f"The deep scrub time stamp before deep-scrub -{init_deep_scrub_stamp}"
+        )
+        rados_object.run_deep_scrub(pgid=pg_id)
+        # wait for deep-scrub to complete in  40 minutes
+        deep_scrub_time = 40
+        if is_deep_scrub_complete(
+            rados_object, pg_id, init_deep_scrub_stamp, deep_scrub_time
+        ):
+            log.error(
+                f"Cannot able to reproduce the issue after waiting the {deep_scrub_time} time"
+            )
+            return 1
+        log.info("The bug is reproduced and verifying the bug fix")
+
+        mon_obj.set_config(
+            section="osd",
+            name="osd_mclock_force_run_benchmark_on_init",
+            value="true",
+        )
+        for osd_id in acting_set:
+            section_id = f"osd.{osd_id}"
+            mon_obj.remove_config(
+                section=section_id, name="osd_mclock_max_capacity_iops_hdd"
+            )
+            rados_object.change_osd_state(action="restart", target=osd_id)
+            time.sleep(5)
+            # Verification of the bug
+            capacity_iops_value = mon_obj.show_config(
+                daemon="osd", id=osd_id, param="osd_mclock_max_capacity_iops_hdd"
+            )
+            log.info(
+                f"The osd.{osd_id} value after removing the parameter -{capacity_iops_value}"
+            )
+            time.sleep(60)
+
+        init_pool_pg_dump = rados_object.get_ceph_pg_dump(pg_id=pg_id)
+        init_deep_scrub_stamp = datetime.datetime.strptime(
+            init_pool_pg_dump["last_deep_scrub_stamp"], "%Y-%m-%dT%H:%M:%S.%f%z"
+        )
+        deep_scrub_time = 20
+        rados_object.run_deep_scrub(pgid=pg_id)
+        deep_result = is_deep_scrub_complete(
+            rados_object, pg_id, init_deep_scrub_stamp, deep_scrub_time
+        )
+        if not deep_result:
+            log.error(f"The deep-scrub not completed within {deep_scrub_time} minutes")
+            return 1
+        # Verification of the osd_mclock_max_capacity_iops_hdd values
+        for osd_id in acting_set:
+            capacity_iops_value = mon_obj.show_config(
+                daemon="osd", id=osd_id, param="osd_mclock_max_capacity_iops_hdd"
+            )
+            if float(capacity_iops_value) < 50 or float(capacity_iops_value) > 500:
+                log.error(
+                    f"The osd_mclock_max_capacity_iops_hdd parameter value for the osd.{osd_id} is "
+                    f"{capacity_iops_value}. The value range is in between 50 and 500"
+                )
+                return 1
```

Comment:
Can this check be moved in the previous for loop?

Reply:
I modified the code.